### PR TITLE
Add German translation

### PR DESCRIPTION
**Pull request by cyb3rko**
You really helped me with your repo so I also help you / this repo by implementing German translations for the licences and the README.

I hope everything works!

### CODE REVIEW

**Comment by santisoler** [comment]
Hi @Cyb3rKo!

I'm really glad this repo turned out to be helpful for other people. I started it a few years ago so I can easily add CC Licenses to my other repositories, but somewhat it reached out and served the same purpose for other people as well.

Your contribution is really interesting! I would be glad to merge it! In fact it's giving me the idea to add a Spanish translation (which is my native language).

I have a question... When I add the CC LICENSE file, GitHub automatically recognizes the license and flags the repo with it (see https://github.com/santisoler/lapis2019 for example, under the _About_ section). Do you know if GitHub still recognizes the German translation?
If not, we should probably encourage people to add both LICENSE files, the one in English and the one with the desired translation. What do you think?

I'll leave a brief review of the PR so we can merge this!

Thanks again for the contribution, I really appreciate that you took the time to do it and opening the PR!

**Comment by cyb3rko** [comment]
Thanks for your quick answer.

I suppose that GitHub just links the LICENSE file in the About section. I don't think we should translate the name of the file.
In my own repo I simply named the file LICENSE but the content is German.

Does that answer your question?

**Review by santisoler — CHANGES_REQUESTED** [review]
What if instead of having a Translations section at the bottom, we have some badges under the `README.md` title for each language?

Something like:
```
[![Deutsche](https://img.shields.io/badge/language-deutsche-red)](https://github.com/santisoler/cc-licenses/tree/master/[DE] German)
```
What will look like this: 

[Deutsche](https://github.com/santisoler/cc-licenses/tree/master/[DE]%20German)

Therefore, clicking the badge with lead directly to the translated `README.md`.

Applying this change may leave the Table of Contents a little bit unnecessary. What do you think?

I would also change the translation directory name, maybe to just `de` if you like.
So, for the Spanish translation, the directory would be `es`.

Feel free to disagree! Although this repo is in my account, your contribution is entirely yours, so feel free to decide weather the changes I'm proposing are good or not.

**Comment by santisoler** [comment]
> Thanks for your quick answer.
> 
> I suppose that GitHub just links the LICENSE file in the About section. I don't think we should translate the name of the file.
> In my own repo I simply named the file LICENSE but the content is German.
> 
> Does that answer your question?

I'm not sure. You're right that GitHub recognizes that there's a license file on the repository, but if you check the repo I linked before, you can see it actually recognizes the LICENSE: CC-BY. In fact, if you click on it, it will show you the LICENSE file (as you mentioned), but it also adds a brief explanation of the license terms: https://github.com/santisoler/lapis2019/blob/master/LICENSE

But this is just a curiosity. If Github doesn't recognize the translation, no problem, we can still merge the PR and maybe add a note to add both the English and the translated version of the LICENSE if you want Github to actually recognize it.

**Comment by cyb3rko** [comment]
> What if instead of having a Translations section at the bottom, we have some badges under the `README.md` title for each language?
> ...
> Therefore, clicking the badge with lead directly to the translated `README.md`.
> 
> Applying this change may leave the Table of Contents a little bit unnecessary. What do you think?
> ...
> Feel free to disagree! Although this repo is in my account, your contribution is entirely yours, so feel free to decide weather the changes I'm proposing are good or not.

I think this is a good idea.
Then I would slightly change the badge to be like this:

`[Deutsch](https://github.com/santisoler/cc-licenses/tree/master/[DE] German)`

[Deutsch](https://github.com/santisoler/cc-licenses/tree/master/[DE]%20German)

I think it's a bit more clean. If you have multiple translations, it could be easier for everybody.

Then I also would go with your other ideas and rename the folder and restructure the README (removing the table of contents, removing translation chapter ...).

I will include this in a few hours, first I will do something else.

Greetings!

**Comment by santisoler** [comment]
Awesome! There's no hurry, so take your time.

BTW, I've just seen you made the changes on your `master` branch. It's not a big problem now, but after I merge this PR into my `master`, your `master` will diverge from mine. A solution to this is to make PRs from branches other than `master`. So, after you fork a repo for contributing you don't commit anything to `master`, you open branches for making changes.

**Comment by cyb3rko** [comment]
Ok, I will go into it now.

Oh, I see. I've just used forking once so I have simply no experience with that.
The last time I just deleted the fork after it was merged because GitHub suggested that.
Maybe deleting it makes it easier because if you don't delete it, the list of repositories will get longer and longer.

**Comment by cyb3rko** [comment]
> I have a question... When I add the CC LICENSE file, GitHub automatically recognizes the license and flags the repo with it (see https://github.com/santisoler/lapis2019 for example, under the _About_ section). Do you know if GitHub still recognizes the German translation?

I tested it on a test repo. My translated README is not recognized so there`s no short version of GitHub given.

I think it's not a big problem so I just added the translated version to my own repo, but if you want to handle it the way you suggested (including two LICENSE files), do it so! Then I can directly add it to this pull request.

**Comment by santisoler** [comment]
> Oh, I see. I've just used forking once so I have simply no experience with that.
> The last time I just deleted the fork after it was merged because GitHub suggested that.
> Maybe deleting it makes it easier because if you don't delete it, the list of repositories will get longer and longer.

No problem! The first time forking and opening PRs is challenging. The good thing about git and GitHub is that it's very unlikely you can break things which aren't easy to solve.

Yep, probably the GitHub suggestion is the easier one. But deleting forks and creating new ones each time a PR of your authorship is merged it's super annoying. In fact, it's better to leave the fork if you might contribute to the original repo in the future, there's no problem with having a long list of repositories. Aaaand, in case you might want to open two PRs, you can do it within the same fork, just from different branches.

**Comment by santisoler** [comment]
> I tested it on a test repo. My translated README is not recognized so there`s no short version of GitHub given.

> I think it's not a big problem so I just added the translated version to my own repo, but if you want to handle it the way you suggested (including two LICENSE files), do it so! Then I can directly add it to this pull request.

Thanks for checking! I suspected that GitHub would only recognize English licenses.
Don't mind about this, just finish your edits and then we will merge it. I might add this advice later on the English `README.md`, and, if you are ok with it, I'll ask you if you can translate it to German on a different PR.

Thanks again for the idea of adding translations. I have already opened an issue for adding the Spanish translation!

**Comment by cyb3rko** [comment]
Okay!

Did you take a look at my new commits?

File: README.md [hunk 1]
```diff
@@ -1,42 +1,39 @@
 # Creative Commons Licenses for GitHub Projects
 
-In this repo you can find easy ways for applying Creative Commons Licenses on
-Github repositories through Markdown language.
+[![Deutsch](https://img.shields.io/badge/translation-DE-red)](de/README.md)
```
Comment (santisoler) [inline README.md]: Nice badge! I would point towards the `de` directory instead to the `README.md` itself. The readme will be shown, but also the LICENSE files.
```suggestion
[![Deutsch](https://img.shields.io/badge/translation-DE-red)](de/)
```

File: de/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Creative Commons Licenses für GitHub-Projekte
+
```
Comment (santisoler) [inline de/README.md:2]: I would add a badge for returning to the English version. Do you agree?

**Review by santisoler — CHANGES_REQUESTED** [review]
This is looking great! I left two minor comments about badges... check them out and I think it will be ready to be merged.

File: README.md [hunk 3]
```diff
@@ -1,42 +1,39 @@
...
+
+---
+
+In this repo you can find easy ways for applying Creative Commons Licenses on Github repositories through Markdown language.
 
 > **WARNING:**
-> Your shouldn't use Creative Commons licenses for software.
+> You shouldn't use Creative Commons licenses for software.
```
Comment (santisoler) [inline README.md:10]: Nice catch!

File: de/README.md (hunk 2 above)
Comment (cyb3rko) [reply de/README.md:2]: Yep, also thought about it, I'll include it!

File: README.md (hunk 3 above)
Comment (cyb3rko) [reply README.md:10]: :)

File: README.md (hunk 1 above)
Comment (cyb3rko) [reply README.md]: Also thought about it but you wrote a few comments before that you want to link to READMEs.

**Review by santisoler — APPROVED** [review]

**Comment by santisoler** [comment]
Thanks a lot @cyb3rko for this contribution! I really appreciate it! :tada: :tada:

**Comment by cyb3rko** [comment]
Sorry, I tried to fix my forked repo and to add a new branch but it completely failed.
Next time I directly try to use multiple branches.

**Comment by santisoler** [comment]
> Sorry, I tried to fix my forked repo and to add a new branch but it completely failed.
> Next time I directly try to use multiple branches.

No problem. Solving that particular situation is not easy for beginners. You have to mess around with hard resets and that's one of the tricks that **can** break things!
It's much easier to delete the fork, fork it again and learn how to manage branches before opening PRs!

**Pull request merged**